instrumentation-genai/opentelemetry-instrumentation-vertexai-v2/src/opentelemetry/instrumentation/vertexai_v2/__init__.py

```diff
@@ -0,0 +1,450 @@
+    assistant_event,
+    user_event,
+)
+from opentelemetry.instrumentation.vertexai_v2.utils import dont_throw
+from opentelemetry.instrumentation.vertexai_v2.version import __version__
+from opentelemetry.semconv._incubating.attributes import gen_ai_attributes
+from opentelemetry.trace import SpanKind, TracerProvider, get_tracer
+from opentelemetry.trace.status import Status, StatusCode
+
+logger = logging.getLogger(__name__)
+
+_instruments = ("google-cloud-aiplatform >= 1.38.1",)
+
+# TODO: span_name should no longer be needed as it comes from `{gen_ai.operation.name} {gen_ai.request.model}`
+WRAPPED_METHODS = [
+    {
+        "package": "vertexai.generative_models",
+        "object": "GenerativeModel",
+        "method": "generate_content",
+        "span_name": "vertexai.generate_content",
+        "is_async": False,
+    },
+    {
+        "package": "vertexai.generative_models",
+        "object": "GenerativeModel",
+        "method": "generate_content_async",
+        "span_name": "vertexai.generate_content_async",
+        "is_async": True,
+    },
+    {
+        "package": "vertexai.preview.generative_models",
+        "object": "GenerativeModel",
+        "method": "generate_content",
+        "span_name": "vertexai.generate_content",
+        "is_async": False,
+    },
+    {
+        "package": "vertexai.preview.generative_models",
+        "object": "GenerativeModel",
+        "method": "generate_content_async",
+        "span_name": "vertexai.generate_content_async",
+        "is_async": True,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict",
+        "span_name": "vertexai.predict",
+        "is_async": False,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_async",
+        "span_name": "vertexai.predict_async",
+        "is_async": True,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_streaming",
+        "span_name": "vertexai.predict_streaming",
+        "is_async": False,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_streaming_async",
+        "span_name": "vertexai.predict_streaming_async",
+        "is_async": True,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "ChatSession",
+        "method": "send_message",
+        "span_name": "vertexai.send_message",
+        "is_async": False,
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "ChatSession",
+        "method": "send_message_streaming",
+        "span_name": "vertexai.send_message_streaming",
+        "is_async": False,
+    },
+]
+
+
+def should_send_prompts():
+    # Previously was opt-in by the following check for privacy reasons:
```

Comment:
openai is using parameter as this to control https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation-genai/opentelemetry-instrumentation-openai-v2/src/opentelemetry/instrumentation/openai_v2/utils.py#L39-L42

Reply:
@aabmass FYI, @lzchen is building some common utiles to help this https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3081/files#diff-cebc2e35b4637202a2226845d87b5093dff75151a31c30000201327be64e6cf0